### PR TITLE
Add Config Flow to Obihai

### PR DESCRIPTION
**Pull request by ejpenney**
## Proposed change

Adding config_flow to the Obihai integration, deprecate YAML configuration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/88627
- Link to parent pull request in the Brands repository: N/A
- This PR fixes or closes issue: fixes: N/A

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by ejpenney** [comment]
Ended up re-creating the PR because I wanted a more descriptive branch name.  Gonna have 3 docs PRs related to this feature, and who knows how long they'll take to get through, so this way future-me can keep better track.

**Comment by ejpenney** [comment]
@dshokouhi, the code was merged. Could I pester you for a +1 here?

Edit: Not sure why asking the codeowner for a review is abuse?  Is there a different process?

**Comment by dshokouhi** [comment]
@ejpenney  I do not usually review PRs in this repo and I think for Config Flow PRs there may need to be a flag at the top to indicate a config flow is added

`ha_config_flow: true`

I am also not sure if the docs should continue to mention the YAML configuration since its being deprecated.

**Comment by ejpenney** [comment]
I don't think it's still mentioning the YAML config?  We still describe the three required input fields, but it's not in the context of YAML anymore.  Unless you see something I don't.

Anyways, I guess I don't understand the process here.  Who should give the PR a review then?  The automation seems to believe it should be codeowners, but I see that's not the case.

File: source/_integrations/obihai.markdown [hunk 1]
```diff
@@ -15,16 +16,9 @@ ha_integration_type: integration
...
-sensor:
-  - platform: obihai
-    host: 192.168.1.x
-```
-
-{% configuration %}
+{% configuration_basic %}
 host:
   description: IP Address of Obihai device
   required: true
```
Comment (frenck) [inline source/_integrations/obihai.markdown]: Configuration basic only has a `description` property on each of its items.
All others are not used or viable for use and should be removed.

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — COMMENTED** [review]
Thanks, @ejpenney 👍

../Frenck

**Review by frenck — APPROVED** [review]

**Pull request merged**